### ISSUE TITLE
components.d.ts does not update when component file added/deleted

### ISSUE THREAD

**Issue by zhuscat** [antfu/unplugin-vue-components#652]
### Describe the bug

`components.d.ts` does not update when file added/deleted in `src/components`(using vue cli, `pnpm serve` is running).

I'm also using Nuxt, I noticed that in Nuxt when file created/deleted in components directory, `components.d.ts` will update. So I'm wondering if this is a BUG.

### Reproduction

https://github.com/zhuscat/test-unplugin-vue-components

### System Info

```Shell
System:
    OS: macOS 13.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 2.91 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
    pnpm: 8.6.1 - ~/.nvm/versions/node/v16.15.0/bin/pnpm
  Browsers:
    Chrome: 114.0.5735.198
    Safari: 16.4
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lanpangzhi** [comment]
vite 里也有这个问题

**Comment by lishaobos** [comment]
我测试中新增是会自动添加的，删除没有，删除这个可以优化

**Comment by memen45** [comment]
Same issue here. Imports are added, but not removed. Is there a way to manually remove imports from / update the components file?

**Comment by l1548844035** [comment]
而且在vite里面你重命名组件也只是新增了重命名组件后的文件，不是直接修改旧的或者将旧的删除，还是有什么参数没加吗

**Comment by tassin-gauthier** [comment]
I agree, addition of component update the components.d.ts file, but deletion of component doesn't. 

The current solution is to let the wrong components into the components.d.ts file, remove manually the components from the file, or delete the components.d.ts and recreate it. Not really ergonomic, and I think it could be easy to detect this change.

**Comment by digisomni** [comment]
Yeah this is also an issue for me, should auto delete the old stuff.

**Comment by Kasopej** [comment]
any update on this issue? I currently have to restart my dev server every time I add a new component, really painful!
Thanks in advance!

**Comment by narduin** [comment]
Hello, this is still an issue today.
Only workaround is either to build the project or to manually delete the file, which is not ideal.
